### PR TITLE
RavenDB-16082 About View: Display connectivity to api.ravendb.net

### PR DESCRIPTION
**Pull request by Danielle9897**
### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-16082

### Additional description
Show License Connectivity on the About page + can refresh

### Type of change
- New feature

### How risky is the change?
- Low 

### Backward compatibility
- Non breaking change

### Is it platform specific issue?
- No

### Documentation update
- This change requires a documentation update. 

### Testing 
- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?
- No

### UI work
- No UI work is needed

### CODE REVIEW

File: src/Raven.Studio/typescript/commands/licensing/getConnectivityToLicenseServerCommand.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿import commandBase = require("commands/commandBase");
+import endpoints = require("endpoints");
+
+class getConnectivityToLicenseServerCommand extends commandBase {
+ 
+    execute(): JQueryPromise<Raven.Server.Web.Studio.LicenseHandler.ConnectivityToLicenseServer> {
+        const url = endpoints.global.license.licenseServerConnectivity;
+
+        return this.query<any>(url, null)
```
Comment (ml054) [inline src/Raven.Studio/typescript/commands/licensing/getConnectivityToLicenseServerCommand.ts:9]: don't we need to pass datatype: undefined? IIRC endpoint results empty result. Please check if it doens't return empty string along with header: `applicatoin/json`.

File: src/Raven.Studio/typescript/commands/licensing/getConnectivityToLicenseServerCommand.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+            .fail((response: JQueryXHR) => {
+                this.reportError("Failed to get connectivity to license server", response.responseText, response.statusText);
```
Comment (ml054) [inline src/Raven.Studio/typescript/commands/licensing/getConnectivityToLicenseServerCommand.ts]: Failed to connect to license server?

File: src/Raven.Studio/typescript/viewmodels/shell/about.ts [hunk 3]
```diff
@@ -339,9 +344,41 @@ class about extends viewModelBase {
         shell.openFeedbackForm();
     }
 
+    private getConnectivityToLicenseServer() {
+        this.spinners.connectivityToLicenseServer(true);
+        return new getConnectivityToLicenseServerCommand()
+            .execute()
+            .done((connectionResult: Raven.Server.Web.Studio.LicenseHandler.ConnectivityToLicenseServer) =>
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/shell/about.ts]: yes - you missed datatype: undefined ;-)

File: src/Raven.Studio/typescript/viewmodels/shell/about.ts [hunk 4]
```diff
@@ -339,9 +344,41 @@ class about extends viewModelBase {
...
+                this.connectedToLicenseServer(connectionResult.StatusCode !== "OK"))
+            .always(() => this.spinners.connectivityToLicenseServer(false));
+    }
+
+    refreshConnectivityToLicenseServer() {
+        this.spinners.connectivityToLicenseServer(true);
+        const cmd = new getConnectivityToLicenseServerCommand();
+        cmd.execute()
+            .done((connectionResult: Raven.Server.Web.Studio.LicenseHandler.ConnectivityToLicenseServer) =>
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/shell/about.ts]: same here.

File: src/Raven.Studio/typescript/viewmodels/shell/about.ts (hunk 3 above)
Comment (ml054) [reply src/Raven.Studio/typescript/viewmodels/shell/about.ts]: done means everything is fine

File: src/Raven.Studio/typescript/viewmodels/shell/about.ts [hunk 5]
```diff
@@ -339,9 +344,41 @@ class about extends viewModelBase {
...
+    private getConnectivityToLicenseServer() {
+        this.spinners.connectivityToLicenseServer(true);
+        return new getConnectivityToLicenseServerCommand()
+            .execute()
+            .done((connectionResult: Raven.Server.Web.Studio.LicenseHandler.ConnectivityToLicenseServer) =>
+                this.connectedToLicenseServer(connectionResult.StatusCode !== "OK"))
+            .always(() => this.spinners.connectivityToLicenseServer(false));
+    }
+
+    refreshConnectivityToLicenseServer() {
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/shell/about.ts]: why we need both getConnectivityToLicenseServer and refreshConnectivityToLicenseServer

File: src/Raven.Studio/typescript/viewmodels/shell/about.ts [hunk 6]
```diff
@@ -339,9 +344,41 @@ class about extends viewModelBase {
...
+                           this.getConnectivityToLicenseServer();
+    }
+    
+    compositionComplete() {
+        super.compositionComplete();
+        
+        popoverUtils.longWithHover($(".not-connected"),
+            {
+                content:
+                    `<small>Allow your server access to <code>api.ravendb.net</code><br>or contact support for an updated license.</small>`,
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/shell/about.ts]: there might be different causes then access - (slow network , another issues)

File: src/Raven.Studio/wwwroot/App/views/shell/about.html [hunk 7]
```diff
@@ -75,13 +75,24 @@ <h2>License information</h2>
...
+                            <div class="big-label margin-right margin-right-sm" data-bind="visible: formattedExpiration">
                                 <div class="content">
                                     <small data-bind="text: expiresText"></small>
                                     <strong data-bind="html: formattedExpiration"></strong>
                                     <small data-bind="text: automaticRenewText"></small>
                                 </div>
                             </div>
+                            <div class="big-label">
+                                <div class="content">
+                                    <small>License Connectivity
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/shell/about.html]: I dont quite like this name. We don't connect license - we check if license server can be reached.

Comment (ml054) [reply src/Raven.Studio/wwwroot/App/views/shell/about.html]: image

Comment (ml054) [reply src/Raven.Studio/wwwroot/App/views/shell/about.html]: image

Comment (ml054) [reply src/Raven.Studio/wwwroot/App/views/shell/about.html]: Maybe let's call this: `License Server` with states:
No connection (red with icon + tooltip)
Valid connection - green , w/o tooltip?

File: src/Raven.Studio/typescript/commands/licensing/getConnectivityToLicenseServerCommand.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/commands/licensing/getConnectivityToLicenseServerCommand.ts]: done
`No connection to RavenDB License Server`

File: src/Raven.Studio/wwwroot/Content/css/themes/blue-theme-variables-override.less [hunk 8]
```diff
@@ -49,3 +49,6 @@
 @token-string: darken(@color-3-1, 15%);
 
 @badge-bg: @gray-dark;
+
+@connected-color: @brand-success;
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/Content/css/themes/blue-theme-variables-override.less]: avoid customization here - use brand-warning

File: src/Raven.Studio/typescript/viewmodels/shell/about.ts (hunk 5 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/shell/about.ts]: no need - removed the 'refresh'

File: src/Raven.Studio/typescript/viewmodels/shell/about.ts (hunk 6 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/shell/about.ts]: =>
Unable to reach the RavenDB License Server at <code>api.ravendb.net</code>

File: src/Raven.Studio/wwwroot/Content/css/themes/blue-theme-variables-override.less (hunk 8 above)
Comment (Danielle9897) [reply src/Raven.Studio/wwwroot/Content/css/themes/blue-theme-variables-override.less]: done

File: src/Raven.Studio/wwwroot/App/views/shell/about.html (hunk 7 above)
Comment (Danielle9897) [reply src/Raven.Studio/wwwroot/App/views/shell/about.html]: done

File: src/Raven.Studio/typescript/commands/licensing/getConnectivityToLicenseServerCommand.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/commands/licensing/getConnectivityToLicenseServerCommand.ts:9]: leaving this - as discussed

**Review by ml054 — APPROVED** [review]

**Pull request merged**